Prof stef: move to newtools

un-mec-paumer opened the pull request:
hello @MarcusDenker I finish every thing in this pull request if you need some change tell when you want

Commit:
add of new view for add/modif a binding

Commit:
Changed layout to make it easier to add and edit bindings with a button.

Commit:
finish of the add / modif binding

Commit:
add of all code for profStef

Commit:
some change to keep the experimental part out of the pull request

jecisc commented:
Hi,

Ideally a PR should be for 1 change. 
So ideally we should have one PR for the playground bindings improvements and one for Prof Stef migration. 
Because 3500 lines of changes is huge and it's hard to review a PR touching multiple things

jecisc commented on src/NewTools-Playground-Tests/StPlaygroundBindingsPresenterTest.class.st:
```diff
@@ -0,0 +1,154 @@
+{ #category : 'accessing' }
+StPlaygroundBindingsPresenterTest >> classToTest [
+
+	^ StPlaygroundBindingsPresenter.
```
No need to add points at the end of a method. Don't hesitate to do CMD/Ctrl + Shift + f on your methods to format them.

```suggestion
	^ StPlaygroundBindingsPresenter
```

jecisc commented on src/NewTools-Playground-Tests/StPlaygroundBindingsPresenterTest.class.st:
```diff
@@ -0,0 +1,154 @@
+{ #category : 'tests' }
+StPlaygroundBindingsPresenterTest >> testInit [
+
+	self assert: presenter notNil.
```
We tend to use #isNotNil instead of #notNil
```suggestion
	self assert: presenter isNotNil.
```

jecisc commented on src/NewTools-Playground-Tests/StPlaygroundPagePresenterTest.class.st:
```diff
@@ -59,6 +59,17 @@ StPlaygroundPagePresenterTest >> testAddToolbarCommandsTo [
+	string := 'coucou'.
+	self openInstance.
+	
+	self assert: presenter text text equals:''.
```
`presenter text text` would not make demeter happy 

(Demeter's law state that we should not show the internal implementation)

jecisc commented on src/NewTools-Playground-Tests/StPlaygroundPageTest.class.st:
```diff
@@ -76,6 +117,53 @@ StPlaygroundPageTest >> testFromReference [
+	|page|
+	page := self newPage .
+	
+	self assert: page isEmpty.
```
FYI we can use #assertEmpty: and #denyEmpty:

The advantage is that in the case the test fails on the CI, the log will display what "page" contains to help debug.

I'm suggestion one rewrite but I've seen other assertions like this in this PR

```suggestion
	self assertEmpty: page.
```

jecisc commented on src/NewTools-Playground-Tests/StPlaygroundPageTest.class.st:
```diff
@@ -76,6 +117,53 @@ StPlaygroundPageTest >> testFromReference [
+	page flush.
+	
+	self assert: (self dateAroundToSecond: page modificationTime) equals: (self dateAroundToSecond: DateAndTime now).
+	
+	
+	
+	
+	
+]
```
This would also be prevented by a formating :)

jecisc commented on src/NewTools-Playground/StPlaygroundBindingAdd.class.st:
```diff
@@ -0,0 +1,34 @@
+StPlaygroundBindingAdd >> buttonAction [
+	|test|	
+		
+		bindingName text isEmpty ifTrue: [ self error: 'I need a string to by identify' . ^ nil ].
+		bindingValue text isEmpty ifTrue: [ self error: 'I need a value' . ^ nil  ].
```
```suggestion
		bindingName text ifEmpty: [ self error: 'I need a string to by identify' . ^ nil ].
		bindingValue text ifEmpty: [ self error: 'I need a value' . ^ nil  ].
```

jecisc commented on src/NewTools-Playground/StPlaygroundBindingAdd.class.st:
```diff
@@ -0,0 +1,34 @@
+
+{ #category : 'accessing' }
+StPlaygroundBindingAdd >> nameButton [ 
+	^ 'add'
```
I did not checkout your PR in an image so I don't know it it's automatically captialized, but if it is not, it should be.

jecisc commented on src/NewTools-Playground/StPlaygroundBindingAdd.class.st:
```diff
@@ -0,0 +1,34 @@
+
+{ #category : 'accessing' }
+StPlaygroundBindingAdd >> windowTitle [ 
+	^ 'new binding'
```
Same, capitalization

jecisc commented on src/NewTools-Playground/StPlaygroundBindingModif.class.st:
```diff
@@ -0,0 +1,59 @@
+
+{ #category : 'accessing' }
+StPlaygroundBindingModif class >> model: anModelPresenter [
+	^ self basicNew 
```
Can't we call #on: and set all of this in #setModelBeforeInitialization: since all parameters are set using anModelPresenter.

Also, aModelPresenter and not anModelPresenter

jecisc commented on src/NewTools-Playground/StPlaygroundBindingModif.class.st:
```diff
@@ -0,0 +1,59 @@
+
+{ #category : 'accessing' }
+StPlaygroundBindingModif >> nameButton [
+	^ 'modif'
```
'modif' is not a good name

jecisc commented on src/NewTools-Playground/StPlaygroundBindingModif.class.st:
```diff
@@ -0,0 +1,59 @@
+Class {
+	#name : 'StPlaygroundBindingModif',
```
In Pharo we want to avoid abbreviation

jecisc commented on src/NewTools-Playground/StPlaygroundBindingPresenter.class.st:
```diff
@@ -0,0 +1,102 @@
+
+{ #category : 'accessing' }
+StPlaygroundBindingPresenter class >> model: anModelPresenter [
+	^ self basicNew 
+		interactionModel: anModelPresenter parent interactionModel;
+		setModelBeforeInitialization: anModelPresenter;
+		initialize
```
Same as comment above

jecisc commented on src/NewTools-Playground/StPlaygroundBindingPresenter.class.st:
```diff
@@ -0,0 +1,102 @@
+			add: (SpBoxLayout new
+				   add: (SpBoxLayout newLeftToRight hAlignCenter add: labelName ; yourself) ;
+				   add: bindingName; 
+				   yourself) withConstraints: [:contrain | contrain width: 75; height: 60; padding: 10];
+		 	add: bindingValue;
+			add: (SpBoxLayout newLeftToRight hAlignCenter ;
+						add: button; 
+						add: buttonClose;
+						yourself) withConstraints: [ :contrain | contrain width: 70; height: 30 ];
+			
+		   yourself
```
Ideally, #withContraints: should be avoided. Check with Esteban if we can avoid it here

jecisc commented on src/NewTools-Playground/StPlaygroundPagePresenter.class.st:
```diff
@@ -191,9 +192,10 @@ StPlaygroundPagePresenter >> hasOutputActivationPort [
-
-	loading := false.
-	super initialize
+   loading := false.
+   super initialize.
```
super should be the first call of this method

un-mec-paumer replied:
i not but if you change the order you got an error like that 
![image_2025-07-02_161753994](https://github.com/user-attachments/assets/6b03ba09-6ad1-469f-9adc-c6ddf609edc2)

jecisc commented on src/ProfStef-Core/Lesson.class.st:
```diff
@@ -0,0 +1,50 @@
+Lesson >> printOn: aStream [
+	super printOn: aStream.
+	aStream
+		nextPutAll: '(';
```
When there is only one character we can use `s nextPut: $(`

jecisc commented on src/ProfStef-Core/LessonView.class.st:
```diff
@@ -0,0 +1,106 @@
+	(aBuilder item: #'Do it' translated)
+		keyText: 'd';
+		selector: #doIt;
+		help: nil;
```
Help to nil is not really good

jecisc commented on src/ProfStef-Core/StProfStefLessonChooser.class.st:
```diff
@@ -0,0 +1,99 @@
+}
+
+{ #category : 'private - accessing' }
+StProfStefLessonChooser class >> model: aModel [
+	^ self basicNew 
+	profStef: ProfStef default;
+	setModelBeforeInitialization: aModel;
+	initialize;
+	yourself.
+]
+
+{ #category : 'layout' }
+StProfStefLessonChooser >> defaultLayout [ 
+	^ (SpBoxLayout newTopToBottom 
+		add: listLesson;
+		add: (SpBoxLayout newLeftToRight hAlignEnd;
+					add: cancelButton;
+					add: okButton;
+				
+					yourself) withConstraints: [ :constraint | constraint width: 90 ; height: 30 ] ;
+				
+	
+	yourself).
+]
+
```
Same comments as above for #new and #withConstraints:

jecisc reviewed (changes requested):
Many of my comments are happening multiple times. I did one or two comments for each issues. But it should be improved in multiple places in the PR

Commit:
move the bindding part into an other branche

Commit:
remove some point :)

Commit:
unuse methode

un-mec-paumer commented:
@jecisc I'am in reserach to don't use withConstraints and i hoe fint it soon

Commit:
add some comments for all classes

Commit:
add a testedTutorial because a hate red hithlight

MarcusDenker committed:
Update src/NewTools-Playground-Tests/StPlaygroundBindingsPresenterTest.class.st

Co-authored-by: CyrilFerlicot <cyril@ferlicot.fr>

MarcusDenker committed:
Update src/NewTools-Playground-Tests/StPlaygroundPageTest.class.st

Co-authored-by: CyrilFerlicot <cyril@ferlicot.fr>

Commit:
testNoUnusedInstanceVariablesLeft (check)
testAndMakeSureSuperSetupIsCalledAsFirstMessageInSetupMethodsOfTestCases (check) 
testNoEmptyPackages (check)

evrey thing is good

Commit:
Merge 643a63f526f3f776a372805e793db36c9ab92ea0

MarcusDenker commented:
There seems to be still playground related code in the PR

un-mec-paumer commented:
a gone resolve it (I just get the origin code because that's the code of the last pull request)

un-mec-paumer committed:
Merge branch 'Pharo14' into profStef

un-mec-paumer commented:
merge conflict is finished

MarcusDenker commented:
- Do not forget to do "Resolve conversation" for the comments above that are fixed
- We need to check to edit the baseline so the package gets actually loaded (this is BaselineOfNewTools >> baseline:)

Commit:
change in the baseline for git

Commit:
Merge 3bad718491712f26c17e90350f0bf8b98cac8d82

un-mec-paumer committed:
I'm so stupid (Fix Baseline of NewTools)

Commit:
I try some thing but I don't know why I don't have some package mentioned in the baseline

MarcusDenker commented:
Reason is that the baseline has been edited in the meantime

We decided to do it in two steps: merge this and then do a PR with the latest baseline 

Next step: this PR should not change the baseline at all

Commit:
good baseline (for now)

MarcusDenker commented:
- The PR removes classes in Newtools-spotter that it should not

Commit:
I don't know how but I lost StEntry so I re put it in the git

un-mec-paumer commented:
my bad (I don't know how I did it but I did) but now it's good and ready to merge
